include/libsupl.rst

```diff
@@ -0,0 +1,153 @@
+
+The following image shows the steps in the libSUPL connection session.
+
+.. figure:: /images/libSUPL_connection_session.png
+      :alt: LibSUPL connection session
+
+LibSUPL architecture
+********************
+
+The libSUPL architecture consists of modules (both internal and external) and API callback functions.
+These are described in the subsequent sections.
+
+Modules
+=======
+
+There are several modules inbuilt into the LibSUPL architecture as shown in the figure below.
+Additionally, the library makes use of the external module :ref:`ossnokalva_asn1_lib`.
+
+.. figure:: /images/libsupl_module_architecture.png
+      :alt: LibSUPL module architecture
+
+Resource initialization and ownership
++++++++++++++++++++++++++++++++++++++
+
+The buffers used in the SUPL communication session, namely the primary and secondary buffers, are owned by the user of the libSUPL library.
+The buffers are characterized by the sizes defined by LIBSUPL_PRI_BUF_SIZE and LIBSUPL_SEC_BUF_SIZE.
+These parameters are defined in the ``supl_session.h`` header file.
+The buffers are passed to libSUPL through the supl_client_init() function.
+
+Communication with the SUPL server must be handled by the application through the read and write callback functions.
+
+SUPL Flow Control
++++++++++++++++++
+
+LibSUPL is initialized by calling the supl_init() function before the first SUPL session.
+The supl_session() function forms the entry point for the SUPL session.
+There is no specific limit on the number of subsequent SUPL sessions that can be executed from this entry point.
+See the documentation on the `SET-Initiated Non-Roaming Successful Case (Proxy mode)`_ for more information on the SUPL session.
+
+The figure below shows the flow of communication in the SUPL session.
+
+.. figure:: /images/supl_session.png
+      :alt: SUPL session
+
+The user of the libSUPL library can request premature exit from the SUPL session by calling the supl_abort() function.
+In practice, the abort flag is checked on every return from the SUPL Read callback function.
+
+.. _ossnokalva_asn1_lib:
+
+OSS Nokalva ASN.1 (Abstract Syntax Notation One) library
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+`OSS Nokalva ASN.1 library`_ is a third-party library, which is used by libSUPL for decoding and encoding SUPL and LPP messages.
+LibSUPL uses the time-optimized version of the library as it produces the smallest code footprint with libSUPL related message definitions.
+
+Buffer Management
++++++++++++++++++
+
+In this library, two separate buffers (the primary buffer and the secondary buffer) are needed to store encoded and decoded formats of the data concurrently.
+The following sequential diagram and steps show the receiving and sending of the SUPL messages.
+
+.. figure:: /images/supl_buffer_seqdgm.png
+      :alt: SUPL session
+
+Receiving
+   The SUPL Read callback function receives the encoded SUPL message.
```

Comment:
Should we mention that this all of the callbacks must be implemented by the user. And it's done like this to let the library make use of the platform specific communication APIs.